### PR TITLE
WIP: Updated adaptors to be based on 4.0.0-SNAPSHOT (Jax-rs2.0)

### PR DESCRIPTION
**Pull request by jadelkhoury**
This branch is based on the latest changes in LyoDesigner 4.0.0-SNAPSHOT ( and hence Jax-rs2.0)

I think the master branch should still be based on the released 2.4.0, but we do need this branch as long as we have not released 4.0.0 - at the least.

---
This change is [Reviewable](https://reviewable.io/reviews/oslc/lyo-adaptor-sample-modelling/24)

### CODE REVIEW

**Comment by jadelkhoury** [comment]
@berezovskyi @jamsden 
Can you please help me with the review of the pom.xml files at the least?

I would like the pom.xml files to be as neat and minimal as possible, since I can then use them as basis to update our instructions on Wiki on how to create servers/clients in general.
Notes: 
1. There is a new dependencyManagement part, which might be a hack to make Swagger 1.5 work together with Jersey 2.27.) Is there a better way to do it?
2. I could not get the code to work on the latest Jersey 2.28. Only 2.27
3. Any thing else i missed?

The 2 adaptors work fine. THey also include client code.

File: adaptor-rm-webapp/pom.xml
Comment (berezovskyi) [inline adaptor-rm-webapp/pom.xml]: what's this?

**Review by berezovskyi — APPROVED** [review]

**Comment by berezovskyi** [comment]
looks good overall, so I have approved. but please see the comments inside.

File: adaptor-rm-webapp/pom.xml
Comment (jadelkhoury) [reply adaptor-rm-webapp/pom.xml]: It was an attempt to reconsile the Jersey dependency version requested by Swagger 1.5 (Jersey 2.25.1), with the latest one we want to use (2.27).
Now changing Jersey version to be 2.25.1.

**Comment by jadelkhoury** [comment]
@berezovskyi Please have a look at my new patch to see if it is what you had in mind.

File: adaptor-rm-webapp/pom.xml
Comment (berezovskyi) [inline adaptor-rm-webapp/pom.xml]: wait, why did we remove the HK2 DI dep?

Comment (jadelkhoury) [reply adaptor-rm-webapp/pom.xml]: The earliest version of jersey-hk2 is 2.26. Code runs fine without it if we use 2.25.
But now I realize we have a bigger problem. We will need CQs if we go for Jersey 2.25, since Jersey was migrated to Eclipse in 2.27.
And that also made me think that we did NOT file a CQ for Swagger.

**Comment by berezovskyi** [comment]
None of those dependencies are shipped with Lyo releases, no need to CQ them.

--
/Andrew
(from phone)

28 mars 2019 kl. 07:51 skrev Jad El-khoury <notifications@github.com<mailto:notifications@github.com>>:


@jadelkhoury commented on this pull request.

________________________________

In adaptor-rm-webapp/pom.xml<https://github.com/OSLC/lyo-adaptor-sample-modelling/pull/24#discussion_r269874273>:

> @@ -88,11 +90,6 @@
                        <artifactId>jersey-container-servlet</artifactId>
                        <version>${version.jersey}</version>
                </dependency>
-               <dependency>


The earliest version of jersey-hk2 is 2.26. Code runs fine without it if we use 2.25.
But now I realize we have a bigger problem. We will need CQs if we go for Jersey 2.25, since Jersey was migrated to Eclipse in 2.27.
And that also made me think that we did NOT file a CQ for Swagger.

File: adaptor-rm-webapp/pom.xml
Comment (jadelkhoury) [reply adaptor-rm-webapp/pom.xml]: so do I have your blessing on this patch?

Comment (berezovskyi) [reply adaptor-rm-webapp/pom.xml]: yes, please go ahead

**Comment by jadelkhoury** [comment]
We will not merge this branch into master until 4.0.0 is released.
Master is based on 2.4.0.

**Comment by berezovskyi** [comment]
Sounds good to me.

--
–Andrew.

2019-03-28  kl. 14:28 skrev Jad El-khoury <notifications@github.com<mailto:notifications@github.com>>:


We will not merge this branch into master until 4.0.0 is released.
Master is based on 2.4.0.

**Comment by berezovskyi** [comment]
client needs to be updated.

**Comment by berezovskyi** [comment]
This branch has diverged from master and has merge conflicts. I have resolved most of them but there is a conflict in the AIRD file, which is super tricky:

image

**Comment by berezovskyi** [comment]
I have merged the AIRD from master in this case.

**Comment by berezovskyi** [comment]
Just ignore it, I was touching the wrong branch.

**Comment by berezovskyi** [comment]
@jadelkhoury I have switched the code to use OSLC4J Client so that the build stops failing but this needs to be done in LyoD: https://github.com/OSLC/lyo-adaptor-sample-modelling/pull/24/commits/dcb169a4edd8e98a7571c416cc40e89bd1df84a6

**Comment by jadelkhoury** [comment]
I am concerned about all changes done on this branch since https://github.com/OSLC/lyo-adaptor-sample-modelling/pull/24/commits/a4011721f26996275881d2642a61d981adf504b8

1. Changes to generated code should be avoided, since the main point of this project is to demonstrated generated code. We need to avoid manual changes (such as those done on the client).
2. Changes to the non-generated parts of the code should be captured in our instructions http://oslc.github.io/developing-oslc-applications/eclipse_lyo/setup-an-oslc-provider-consumer-application.html 
3. I'd be very very careful when merging AIRD files. (that is avoid.) THere is no point merging from master either. Master is meant to be for code modelled/generated from 2.4.0. THis branch is from 4.0.0. Naturally, they are to diverse.

**Comment by berezovskyi** [comment]
Do you want to drag this branch all the way till the 4.0 is released? Or shall we merge it soon?

**Comment by jadelkhoury** [comment]
> Do you want to drag this branch all the way till the 4.0 is released? Or shall we merge it soon?

My thought is that we wait until 4.0 is released.

**Comment by berezovskyi** [comment]
Do you need any help resolving merge conflicts or with anything else in this PR?

**Comment by jadelkhoury** [comment]
> Do you need any help resolving merge conflicts or with anything else in this PR?

Most likely. my thought is that we move this into master, once we release LyoDesigner 4.0.0, which I hope we can do soon.
So first step is to release 4.0.0.

**Comment by berezovskyi** [comment]
I suggest to keep this as a sample JAX-RS 1.1 implementation instead of merging this and then deprecating in favour of OSLC 2020 refimpl

**Comment by jadelkhoury** [comment]
@berezovskyi Your last comment doesn't make sense to me.
Please, let us discuss this repo before we make any chanfges to it.

**Comment by berezovskyi** [comment]
@jadelkhoury I based it on what you wrote actually in: https://github.com/oslc-op/refimpl/issues/21 If we go that route, let's keep this repo as a valid 2.x Lyo sample code and follow your plan to make refimpl repo the Lyo 4.0 / JAX-RS 2.0 sample code.

**Comment by jadelkhoury** [comment]
It will take some time before we do https://github.com/oslc-op/refimpl/issues/21. Until then, this repo is what I need to have.

If you want to do something on this repo, here's what I suggest (so we avoid all confusing once and for all):

We have two branches 2.4.0 & 4.0.0-SNAPSHOT. No master branch.
that is, either rename "master" into 2.4.0, or create a new branch from master to 2.4.0, and then kill the master branch.

**Comment by berezovskyi** [comment]
Ok, my plan is not to touch master (much) especially given that no 2.x lyo releases are planned to come.

I can rename master into main-2.x and this branch into main-4.x (maybe there will be updates to this repo when 4.1 comes out?)

We can then close this PR as there is no reason to merge it anywhere and I can make main-4.x to be the main branch on Github.

**Comment by jadelkhoury** [comment]
Yes to all in last proposal. thanks @berezovskyi

**Comment by berezovskyi** [comment]
@jadelkhoury done. please delete outdated branches from https://github.com/OSLC/lyo-adaptor-sample-modelling/branches/stale if possible

**Pull request closed without merging**
